src/core/src/service_interfaces/StatusHandler.py

```diff
@@ -101,18 +107,21 @@ def reset_assessment_data(self):
         self.__assessment_packages = []
         self.__assessment_errors = []
         self.__assessment_total_error_count = 0
+        self.__truncated_patches = []
+        self.__assessment_tmp_map = {}
 
     def set_package_assessment_status(self, package_names, package_versions, classification="Other", status="Available"):
         """ Externally available method to set assessment status for one or more packages of the **SAME classification and status** """
         self.composite_logger.log_debug("Setting package assessment status in bulk. [Count={0}]".format(str(len(package_names))))
+
         for package_name, package_version in zip(package_names, package_versions):
             patch_already_saved = False
             patch_id = self.__get_patch_id(package_name, package_version)
-            for i in range(0, len(self.__assessment_packages)):
```

Comment:
self.__assessment_packages has a full lifecycle that __assessment_tmp_map is not fully compensating for with this change.

For example, in master, line 604, where __assessment_packages is read from disk: 
self.__assessment_packages = self.__assessment_summary_json['patches']

self.__assessment_tmp_map is not reloaded at the same place. I'd suggest renaming the map to self.__assessment_packages_map, and scanning the code carefully for all usages of self.__assessment_packages, and ensuring all behaviors are preserved with this change.